src/libasr/pass/intrinsic_functions.h

```diff
@@ -3238,9 +3238,9 @@ namespace Modulo {
         end function
         */
         if (is_real(*arg_types[0])) {                
-            body.push_back(al, b.Assignment(result, b.Sub(args[0], b.Mul(args[1], b.i2r_t(Floor::FLOOR(b, b.Div(args[0], args[1]), int32, scope), arg_types[1])))));
+            body.push_back(al, b.Assignment(result, b.Sub(args[0], b.Mul(b.r2r_t(args[1], arg_types[0]) , b.i2r_t(Floor::FLOOR(b, b.Div(args[0], b.r2r_t(args[1], arg_types[0])), int32, scope), arg_types[1])))));
```

Comment:
```suggestion
            body.push_back(al, b.Assignment(result, b.Sub(args[0], b.Mul(b.r2r_t(args[1], arg_types[1]) , b.i2r_t(Floor::FLOOR(b, b.Div(args[0], b.r2r_t(args[1], arg_types[1])), int32, scope), arg_types[1])))));
```

Why cast it to `arg_types[0]`?

Reply:
I see, it is fine.